CTFL v4.0 SK Beta (#8)
istqborg/istqb_ctfl: syllabus-sk/03-continuous-improvement.md

Reviewer: udržovateľnosti -> udržateľnosti

Aj keď sa z veľkej časti používa na odhaľovanie špecifických defektov kódu, možno ju tiež použiť na vyhodnotenie **udržovateľnosti** a bezpečnosti (security). 

Aj keď sa z veľkej časti používa na odhaľovanie špecifických defektov kódu, možno ju tiež použiť na vyhodnotenie **udržateľnosti** a bezpečnosti (security). 

Statické testovanie je možné použiť na meranie kvalitatívnych charakteristík (napr. **udržovateľnosti**) nezávislých na spustení kódu, dynamické testovanie iba na tie, ktoré sú na spustení kódu závislé (napr. výkonnostná efektivita).

Statické testovanie je možné použiť na meranie kvalitatívnych charakteristík (napr. **udržateľnosti**) nezávislých na spustení kódu, dynamické testovanie iba na tie, ktoré sú na spustení kódu závislé (napr. výkonnostná efektivita).

Reviewer: udržovateľnosti -> udržateľnosti

Aj keď sa z veľkej časti používa na odhaľovanie špecifických defektov kódu, možno ju tiež použiť na vyhodnotenie udržovateľnosti a bezpečnosti (security).

Aj keď sa z veľkej časti používa na odhaľovanie špecifických defektov kódu, možno ju tiež použiť na vyhodnotenie udržateľnosti a bezpečnosti (security).

Statické testovanie je možné použiť na meranie kvalitatívnych charakteristík (napr. udržovateľnosti) nezávislých na spustení kódu, dynamické testovanie iba na tie, ktoré sú na spustení kódu závislé (napr. výkonnostná efektivita).

Statické testovanie je možné použiť na meranie kvalitatívnych charakteristík (napr. udržateľnosti) nezávislých na spustení kódu, dynamické testovanie iba na tie, ktoré sú na spustení kódu závislé (napr. výkonnostná efektivita).

* Ako statické, tak aj dynamické testovanie (vrátane analýzy zlyhania) môže viesť k odhaleniu defektu, avšak existujú niektoré typy defektov, ktoré možno nájsť buď len statickým alebo iba dynamickým testovaním.
* Statické testovanie nájde defekty priamo, zatiaľ čo dynamické testovanie hľadá zlyhanie, z ktorých sú príslušné defekty určené následnou analýzou.
* Statické testovanie môže ľahšie odhaľovať defekty ťažko dosiahnuteľné pomocou dynamického testovania alebo také, ktoré sú v kóde v častiach spúšťaných len zriedka.
* Dynamické testovanie je možné použiť iba na spustiteľné pracovné produkty, statické testovanie aj na nespustiteľné.
* Statické testovanie je možné použiť na meranie kvalitatívnych charakteristík (napr. udržovateľnosti) nezávislých na spustení kódu, dynamické testovanie iba na tie, ktoré sú na spustení kódu závislé (napr. výkonnostná fektivita).

Medzi typické defekty odhaliteľné ľahšie a lacnejšie statickým testovaním patria:

* defekty v požiadavkách (napr. nezrovnalosti, nejednoznačnosti, rozpory, opomenutia, nepresnosti a duplicity),
* defekty v návrhu (napr. neefektívne databázové štruktúry, zlá modularita),

> * niektoré typy programovaích defektov (napr. premenné s nedefinovanými hodnotami, nedeklarované premenné, nedosiahnuteľný alebo duplicitný kód, nadmerná zložitosť kódu),

Author: ```suggestion
* niektoré typy programovacích defektov (napr. premenné s nedefinovanými hodnotami, nedeklarované premenné, nedosiahnuteľný alebo duplicitný kód, nadmerná zložitosť kódu),
```

* **Individuálna revízia**. Každý revidujúci reviduje definovaný pracovný produkt individuálne s cieľom posúdiť jeho kvalitu a identifikovať anomálie, odporúčania a otázky pomocou jednej alebo viacerých techník revízie (napr. revízia založená na kontrolnom zozname, revízia podľa scenárov – pozri normu ISO/IEC 20246 [@ISO20246]). Revidujúci zaznamenáva všetky zistené anomálie, odporúčania a otázky.
* **Komunikácia a analýza**. Nie každá anomália identifikovaná počas revízie musí byť nutne defekt. Preto je nutné všetky takéto anomálie analyzovať a prediskutovať a pri každej by sa malo rozhodnúť o jej stave, vlastníctve a požadovaných opatreniach. To sa zvyčajne vykonáva pri revíznej schôdzke, počas ktorej účastníci tiež diskutujú o úrovni kvality každého revidovaného pracovného produktu a aké následné opatrenia sú vyžadované. Na uzatvorenie opatrení môže byť vyžadovaná ďalšia revízia.

> * **Opravy a reportovanie**. Aby bolo možné vykonať nápravu, mala by byť pre každý defekt vytvorený report o defekte. Pri splnení daných výstupných kritérií môže byť pracovný produkt akceptovaný a výsledky revízie sú podané v správe.

Reviewer 2: Aby bolo možné vykonať nápravu, mala by byť pre každý defekt vytvorený report o defekte.
"mala"   nahradit slovickom "mal"

Reviewer 2: ```suggestion
* **Opravy a reportovanie**. Aby bolo možné vykonať nápravu, mal by byť pre každý defekt vytvorený report o defekte. Pri splnení daných výstupných kritérií môže byť pracovný produkt akceptovaný a výsledky revízie sú podané v správe.
```

> ### Roly a zodpovednosti pri revíziách

Reviewer: Rol**y** a zodpovednosti pri revíziách.

Rol**e** a zodpovednosti pri revíziách

Reviewer 3: Rola sa skloňuje podľa vzoru žena, tzn. správne je v Nom. mn.č. "Roly".

Revíziu môžu vykonávať rôzni ľudia a môžu zastávať rôzne roly. Hlavné roly a ich povinnosti sú:

* **Manažér** – rozhoduje o tom, čo má byť revidované a aké zdroje budú využité (vrátane ľudí a času).
* **Autor** – vytvára a opravuje revidovaný pracovný produkt.

> * **Moderátor** (niekedy aj facilitátor) – zaisťuje efektívny priebeh revíznych schôdzok vrátane využitia mediácie. Okrem iného dohliada na dodržanie časového rámca a zabezpečenie komfortného prostredia, v ktorom môže každý slobodne vyjadriť svoj názor.

Author: Originál "... and a safe review environment in which everyone can speak freely"
Slovník: safety > bezpečie
Nechávam na diskusiu v rámci pracovnej skupiny, ktorá preklad tvorila.

> * **Moderátor** (niekedy aj facilitátor) – zaisťuje efektívny priebeh revíznych schôdzok vrátane využitia mediácie. Okrem iného dohliada na dodržanie časového rámca a zabezpečenie komfortného prostredia, v ktorom môže každý slobodne vyjadriť svoj názor.

Author: ```suggestion
* **Moderátor** (niekedy aj facilitátor) – zaisťuje efektívny priebeh revíznych schôdzok vrátane využitia mediácie. Okrem iného dohliada na dodržanie časového rámca a vytvorenie bezpečného prostredia, v ktorom môže každý slobodne vyjadriť svoj názor.
```

* **Zapisovateľ** – zhromažďuje anomálie od revidujúcich a zaznamenáva ďalšie informácie z revízie ako napr. rôzne rozhodnutia alebo nové anomálie zistené počas revízneho stretnutia.
* **Revidujúci** – vykonáva revíziu. Revidujúcim môže byť niekto, kto pracuje na projekte alebo je odborníkom na danú problematiku alebo ktorákoľvek osoba zo zainteresovaných strán.
* **Vedúci revízie** – preberá celkovú zodpovednosť za revíziu (napr. rozhodovanie o tom, kto do nej bude zapojený) a organizáciu toho, kedy a kde sa bude revízia konať.

Popis ďalších možných rolí možno nájsť v norme ISO/IEC 20246 [@ISO20246].

### Typy revízií

> Existuje mnoho typov revízií, od neformálnych až po veľmi formálne. Požadovaná úroveň formálnosti závisí od faktorov ako je použitý SDLC, vyspelosť procesu vývoja, kritickosť a zložitosť revidovaného pracovného produktu, právne alebo regulatórne požiadavky a potreba doloženia záznamov pre prípadný audit. Rovnaký pracovný produkt môže byť revidovaný rôznymi typmi revízií, napr. najprv neformálne a neskôr formálnejšie.

Reviewer: právne alebo **regulatórne** požiadavky a potreba doloženia

 právne alebo **regulárne** požiadavky a potreba doloženia

Medzi bežne používané typy revízií patria:

* **Neformálna revízia**. Neformálne revízie sa neriadia definovaným procesom a nevyžadujú formálny dokumentovaný výstup. Hlavným cieľom je odhaľovanie anomálií.
* **Predvedenie (walkthrough)**. Predvedenie (vedené autorom) môže slúžiť mnohým cieľom ako je hodnotenie kvality a budovanie dôvery v pracovný produkt, vzdelávanie revidujúcich, dosiahnutie dohody, generovanie nových nápadov, motivácia a podpora autora s cieľom zlepšovať pracovný produkt a odhaľovať anomálie. Revidujúci môžu (ale nemusia) pred predvedením vykonať individuálnu revíziu.
* **Technická revízia**. Technickú revíziu vykonávajú odborne kvalifikovaní revidujúci a vedie ju moderátor. Cieľom technickej revízie je primárne dosiahnuť zhodu a urobiť rozhodnutia týkajúce sa nejakého technického problému, ale tiež odhaliť anomálie, vyhodnotiť kvalitu, vybudovať dôveru v pracovný produkt, generovať nové nápady, motivovať autorov a podporiť ich v zlepšovaní.

> * **Inšpekcia**. Vzhľadom na to, že inšpekcie sú najformálnejším typom revízie, riadia sa komplexným všeobecným procesom (pozri <#section:review-process-activities>). Hlavným cieľom je nachádzať maximálny počet anomálií. Ďalšími cieľmi sú hodnotenie kvality, budovanie dôvery v pracovný produkt, motivácia a podpora autorov v zlepšovaní. Sú zhromažďované metriky, ktoré sa používajú na zlepšovanie celého SDLC vrátane samotného procesu inšpekcie. Pri inšpekciách nemôže autor vystupovať ako vedúci revízie alebo zapisovateľ.*

Reviewer 3: > Sú zhromažďované metriky, ktoré sa používajú na zlepšovanie celého SDLC vrátane samotného procesu inšpekcie.

Vetná stavba nie je ideálna.

```suggestion
* **Inšpekcia**. Vzhľadom na to, že inšpekcie sú najformálnejším typom revízie, riadia sa komplexným všeobecným procesom (pozri <#section:review-process-activities>). Hlavným cieľom je nachádzať maximálny počet anomálií. Ďalšími cieľmi sú hodnotenie kvality, budovanie dôvery v pracovný produkt, motivácia a podpora autorov v zlepšovaní. Metriky sú zhromažďované a používané za účelom zlepšovania celého SDLC, vrátane samotného procesu inšpekcie. Pri inšpekciách nemôže autor vystupovať ako vedúci revízie alebo zapisovateľ.*
```

Návrh úpravy vetnej stavby za účelom zlepšenia čitateľnosti.
